### PR TITLE
Add option to generate QR code for entity when inspecting it

### PR DESCRIPTION
**Pull request by samkumar**
If you want I can also add the option to specify how much error correction to put into the QR code, which would affect the size. For now I default to the "Medium" setting.

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -1094,6 +1110,22 @@ func actionInspect(c *cli.Context) error {
 	if pub {
 		pubObjs(topub, cl, c)
 	}
+	if qr {
+		// Generate a QR code for each entity with an available signing key
+		for _, qrd := range(toqrg) {
+			roi := qrd.ro
+			if (roi.GetRONum() == objects.ROEntity) {
```
Comment (immesys) [inline cli.go:1117]: this would be ROEntityWKey

File: cli.go [hunk 2]
```diff
@@ -1094,6 +1110,22 @@ func actionInspect(c *cli.Context) error {
...
+				entity := roi.(*objects.Entity)
+				signingblob := entity.GetSigningBlob()
+				if signingblob != nil {
+					err := qrcode.WriteFile(string([]byte{objects.ROEntityWKey}) + string(signingblob), qrcode.Medium, 256, fmt.Sprintf("%s.png", qrd.name))
+					if err != nil {
+						fmt.Printf("Could not encode QR Code and write to file: %v\n", err)
```
Comment (immesys) [inline cli.go:1125]: you should exit 1 here

File: cli.go [hunk 3]
```diff
@@ -1094,6 +1110,22 @@ func actionInspect(c *cli.Context) error {
...
 	}
+	if qr {
+		// Generate a QR code for each entity with an available signing key
+		for _, qrd := range(toqrg) {
+			roi := qrd.ro
+			if (roi.GetRONum() == objects.ROEntity) {
+				entity := roi.(*objects.Entity)
+				signingblob := entity.GetSigningBlob()
+				if signingblob != nil {
+					err := qrcode.WriteFile(string([]byte{objects.ROEntityWKey}) + string(signingblob), qrcode.Medium, 256, fmt.Sprintf("%s.png", qrd.name))
```
Comment (immesys) [inline cli.go]: I would default to more than 256 pixels, or perhaps choose the resolution based on the size of signingblob

File: cli.go (hunk 1 above)
Comment (samkumar) [inline cli.go:1117]: I thought so too, but the function used to create Entities (https://github.com/immesys/bw2/blob/dev/objects/routing.go#L1607) checks if the RONum is ROEntityWKey, and if it is, sets it to ROEntity. So GetRONum() would still return ROEntity in this case. That's why I make sure that the RONum is ROEntity and make sure that the signing blob isn't nil.

File: cli.go (hunk 3 above)
Comment (samkumar) [inline cli.go]: Sure.

File: cli.go (hunk 2 above)
Comment (samkumar) [inline cli.go:1125]: OK

**Pull request merged**